### ISSUE TITLE
When will Nancy 2 be out of the pre

### ISSUE THREAD

**Issue by bluetentacle** [NancyFx/Nancy#2872]
2.0.0-clinteastwood was last

**Comment by m-barthelemy** [comment]
Hi,

I have the same exact question.
Nancyfx is still fully relevant even with the latest asp.net releases.

It's a bit scary to see that the whole project seems to have slowed down a lot, with this 2.0 pending release that doesn't have public pre/beta releases anymore.

Is there anything that basic users could help with, without having to fully understand the internals of Nancyfx?

Please take no offense, this is precisely because it's a great project that I/we express such questions and concerns.

**Comment by hikeeba** [comment]
I would also like to know what the plans are. This is a great tool and if there is anything that can be done I'd like to know.

**Comment by khellang** [comment]
> Is there anything that basic users could help with, without having to fully understand the internals of Nancyfx?

You could sponsor some work through [the Patreon](https://www.patreon.com/nancyfx)?

**Comment by m-barthelemy** [comment]
> You could sponsor some work through the Patreon?

Done!

**Comment by Jaxelr** [comment]
Likewise,

May i suggest adding the Patreon information on a more noticeable place (maybe the readme file)? Didn't found out until i read this thread.

**Comment by bloritsch** [comment]
What is left to do in Nancy?  I mean what's the point in donating money to a black hole?  Last update was a couple months ago, but it appeared to be really minor.  I just want a solid base to do web services with.  I need the NetCore support of Nancy 2, but I don't need any view support.  I just need something that works out of the box.  Nancy has been answering the mail on that, but I wonder if I should just use Owin directly.

**Comment by sebandres** [comment]
[This Issue](https://github.com/NancyFx/Nancy/issues/2659) had a checklist of what was needed before 2.0.0-d could be released. But it is now closed without all checks being ticked so no idea.

Was actually wondering the same myself about the progress and what was left in the roadmap (to event see if I could contribute).

**Comment by sebandres** [comment]
Cool @thecodejunkie! Looking forward to it. Would that be version 2.0-d or a final 2.0 release??

**Comment by thecodejunkie** [comment]
Most likely final :D

**Comment by sdmimaye** [comment]
I think we can all agree that the 2.0 Release will not gonna happen. I have been looking at this framework for a while now and i guess its no longer actively developed.

**Comment by snavarropino** [comment]
Any news? I am also thinking we will never have Nancy v2 released :-(

**Comment by herrgruebner** [comment]
Any update on Nancy 2.0? Last mention was "My hope is that we can make that happen this or the week after!" about two months ago...

**Comment by williamhbell** [comment]
Hi Andreas, do you have any thoughts about wrapping up the Nancy 2.0 release?  Is there anything that can be done to help? Best regards, Will

**Comment by sebandres** [comment]
I think at this point given the direction that aspnet core 2.x has taken which aligns to some of the reasons of why nancy was born there isn't probably much value in investing into this framework anymore. Specially at the rate that .net core is moving and given the fact that it is also open source.

Perhaps it would be best to just archive this project so that people don't keep asking when v2 is going to be released?

**Comment by pruiz** [comment]
Well, there is people still using Nancy for already existing projects.. so I would favor releasing 2.0 and letting it clear there wont be a 3.0.. this way, the people who trusted the project wont be left alone in the dark.

**Comment by turbo** [comment]
In fact, there are even people using Nancy for new projects. A formal 2.0 release would be nice.

**Comment by herrgruebner** [comment]
Or upgrading their existing projects to net standard/core from framework.
Having to use an old prerelease isn't great.

**Comment by sebandres** [comment]
Yeah I second herrgruebner, also using a prerelease version in a mission
critical system given the frequency of commits might not have been the best
course forward for a project and that is fine because things change.

Just flagging this as a production ready release if there is no confidence
in it (otherwise it would have happened already) might give the wron
impression and encourage more people to fully adopt it which might do more
harm than good if the community is no longer there to actively fix issues.

**Comment by pruiz** [comment]
AFAIK there _is_ an active community asking for a new release. If someone is not willing to use Nancy anymore, he is more than free to do so.

**Comment by grumpydev** [comment]
Andreas and I intend to look at this when we are both back off holiday, there's some things that need updating so they work with the latest bits (again), then we will put a release out.

**Comment by cguedel** [comment]
Any update?

**Comment by KnowMoreIT** [comment]
Yeah this is the first I've seen about Patreon. Unfortunately I've also started converting my projects because I thought this wasn't actively developed anymore. I have the same Razor issue that I posted about in April of 2017 still open.

**Comment by aespinoza** [comment]
Yes, this is sad. I have also started to port my projects. I am currently looking at using Asp.Net Core with [Carter](https://github.com/CarterCommunity/Carter).

**Comment by GusBeare** [comment]
Shame. I love Nancy and have built a web site, a simple CMS and various web services with it. But Nancy lives on at flicka20.com and many sailors out there enjoy the site it delivers!

**Comment by PedreHavenga** [comment]
Also keeping an eye on this as we want to .Net Core soon. Please any update if the 2.0 would get a release or is this dead in the water?

**Comment by nkosi23** [comment]
I think people talking about the death of the project are being a bit harsh. The project has been alive for many years and the last commit was made 4 months ago which is not that old. There is no outstanding critical bug in the last stable release. So maybe we can say that the unpaid volunteers who have done an outstanding work creating Nancy have been too busy lately to ship the .NET Core support feature, however I think it is premature and unnecessary to bury the project and talk about its death. If I was the creator of the project, I would find this kind of feedback rude and ungrateful, as if all the work done and the history of the project do not matter if a hype feature is not shipped in real-time.

Regarding future relevance, Nancyfx is still highly relevant and the experience of developing websites/web services with Nancyfx is fundamentally different than the one offered by the framework shipped with .Net Core. I do not even understand how people can compare the two. There are also all these people using Nancyfx under Mono (which supports platforms, hardware architectures and build tools that .NET Core have no intention to ever support - at least not in the foreseable future). So Nancyfx is more relevant than ever. As far as we are concerned we are starting our new projects with Nancy with peace of mind.

But quite frankly, Nancy is one of the rare open source projects that we'll fork without hesitation even if it went unmaintained, because otherwise we'd simply end up creating the same thing. Nancy is a very elegant and unique solution to the problem it addresses. Thanks to its flexibility, we've built frameworks on top of it in ways that would have been much more difficult with the asp.net/.net core paradigm.

I appreciate and respect the frustration of those needing this feature but I felt it was important to counter-balance the direction this thread was going to so that the core devs can know that there are many people out there who have no intention to jump ship. Please be patient or invest time to submit a pull request if .NET Core support is so important for you, it will be time well invested. Remember that the source code is out there (and it is well written).

Nancy does not strike me as a dead project or a project without a future. The level of activity is the one I would expect for an already mature open-source project not even having corporate sponsors.

**Comment by turbo** [comment]
> But quite frankly, Nancy is one of the rare open source projects that we'll fork without hesitation

I suspect that has been done by many already. Actually, we're currently in the process of doing that. Nancy is easy enough to understand to fork and maintain it because you can choose only to fork the parts you need.

**Comment by snavarropino** [comment]
I also think the better to clearly show current status and plans. If nancy is already stopped (unfortunately as it seems ) there is no value to hide it.

**Comment by JeancarloFontalvo** [comment]
I'm newbie in this wonderful framework, me and my team, wants to know if is it good to start developing for a production project with this version (2.x) or to use the 1.x one?

Thanks for answering... :smile: and thanks for this awesome masterpiece.

**Comment by JeancarloFontalvo** [comment]
@btecu But why? :open_mouth: I think Nancy seems easier to go also it "just works". Cand you provide your facts to go with go directly with .Net Core, It'd be helpful. :smiley:

**Comment by btecu** [comment]
@JeancarloFontalvo ASP.NET Core has _borrowed_ the best ideas from Nancy (which were not really original to Nancy but still). In addition, Nancy hasn't really had anything merged in the last approximately six months. You don't want to use a framework that's not being maintained.

**Comment by Jaxelr** [comment]
Similar alternatives, [Carter](https://github.com/CarterCommunity/Carter) comes to mind, which of course lives atop ASP.Net Core.

**Comment by crmtogether** [comment]
Nancy just works and if you use docker/mono imo you need not worry too much about the project being maintained as you define the environment and don’t have to worry about OS upgrades. Every bit of software has a limited shelf life anyway. The people working on Nancy did this out of their own time and if they’re busy making money now good luck to them. They don’t need to answer to anyone either. 
If you have a bug, well then offer to pay for the fix. The guys or someone might be willing to do this in a timely manner of paid. If you don’t want to pay for it we’ll then it’s not really an issue for you.

**Comment by snavarropino** [comment]
Totally agree, they don'y have to work in Nancy anymore, but at least, they clould crearly state what is happening with this framnework and if there is any plan or not

**Comment by grumpydev** [comment]
Before this turns into an argument, does anyone active on this issue actually want to *help* get this done? It would be good to get a netstandard 2 (or 2.1?) release done, including all the sub projects, but I have (intentionally) done very little with netcore etc. so far, and had nothing but hassle when I have used it, its not something I can just crank out.

**Comment by snavarropino** [comment]
I have some experience with netstandard and .net core so I could help somehow, but I am going to require some guidance...

**Comment by janoserdelyi** [comment]
as an individual i'd be willing to pay some amount per month

**Comment by gudatcomputers** [comment]
I'm already contributing a small monthly amount via Patreon but would be willing to help get this working.  We've come to love Nancy and use it alot.  Specifically how great it is to test as compared to ASP.NET Core.  Being a TDD shop this is super important to me.  Is there a place where I can see what works is remaining for this? Or is it more just get in there and make it work?

**Comment by danielmoncada** [comment]
@khellang @thecodejunkie @jchannon any update regarding a 2.0 production release?

**Comment by PedreHavenga** [comment]
I am also keeping an eye on this thread. We are fully committed using Nancy in our organisation.

**Comment by danielmoncada** [comment]
@PedreHavenga We ended up switching to Carter, which provides Nancy like syntax on top of ASP.NET Core routing. Check it out: https://github.com/CarterCommunity/Carter

**Comment by Dreamwalker666** [comment]
Can a list of issues be associated to this so people know what needs doing / completing for a 2.0 release?  It's about time it got finalised.

**Comment by Jaxelr** [comment]
You mean like this [milestone?](https://github.com/NancyFx/Nancy/pulls?q=is%3Aopen+is%3Apr+milestone%3A2.0-dangermouse)

**Comment by PedreHavenga** [comment]
> Can a list of issues be associated to this so people know what needs doing / completing for a 2.0 release? It's about time it got finalised.

I can maybe get a Dev or two to work on small issues if we can get a list of Tasks/Bugs that needs to be done to get V2 ready for Release. This will be slow going but we can give it a shot.

**Comment by aespinoza** [comment]
At least a year without a response from the maintainers makes it clear Nancy is dead.  My recommendation is to use https://github.com/CarterCommunity/Carter

**Comment by snavarropino** [comment]
I suggested some help from my side several months ago. This help is still on the table in order to release Nancy, in this case working for .net core 3

**Comment by cocowalla** [comment]
> 
> 
> At least a year without a response from the maintainers makes it clear Nancy is dead. My recommendation is to use https://github.com/CarterCommunity/Carter

I've been building web apps with Nancy for years, and I'm genuinely sad to see this - it's certainly not on the "super duper happy path" now 😞 

In years past, ASP.NET just wasn't extensible enough to support a lot of scenarios elegantly, but Nancy was unopinionated and had an extensibility point for every occasion. Now, ASP.NET Core takes extensibility seriously (the fact Carter even exists is testament to that), and is finally in great shape - this has undoubtedly played a large part in the decline in popularity of Nancy, and indeed of the morale of the devs.

So while Nancy has been a prominent and much loved and library for almost a decade, as sad as it is, I can at least understand how it's gotten this way. I would however strongly urge the main devs to make a decision and _comment_ on the current situation, ending a year of uncertainty one way or another.

**Comment by grumpydev** [comment]
I still want to put another release out, probably just of the "core" of Nancy (so no additional container support etc.), but every time I've tried to use .net core I've ended up in a world of pain so I generally avoid it. 

Last time I chatted to @jchannon he mentioned that the main core seemed to build and work ok with the latest bits, if that's still the case I can probably manually package up a release, if that's not the case then I will need someone's help to get a list together of what needs fixing *purely to get it to work (probably) netstandard 2.0*.

**Comment by cocowalla** [comment]
@grumpydev FWIW, I've been using `2.0.0-barneyrubble` with `netstandard2.0` for a long time, without any dotnet-related issues.

**Comment by grumpydev** [comment]
@cocowalla so if we were to just put out the core nancy packages of -barneyrubble out as 2.0 "final" would that be acceptable, at least as a "first step" ?

I still use Nancy every single day, and it still works brilliantly for me, but I don't use .net core for the reasons previously mentioned - I'll return to .net core at some point, but at the moment the amount of friction it gives me just isn't worth my time/frustration, so I need help in .net core/netstandard specific stuff.

**Comment by jchannon** [comment]
So currently our latest is 2.0-clinteastwood and that is stable for my part.  There have been a few minor changes I believe since that went out.

As long as you have .NET and .NETCore installed on a Windows box you should be able to build it fine. I've just tested on OSX for netcore and all was fine. Can't build .NET as I'm on OSX

**Comment by cocowalla** [comment]
@grumpydev @jchannon yes, in the short term what people really want is a production-ready version of 2.0 on NuGet.

Beyond that though, it would be great if you (and/or the other core maintainers) got together and publically said something about the future of Nancy. Is there a roadmap? Will it continue to be maintained? Will 2.0 be considered "completed software", with just bugfixes? So many questions from the community :open_hands:

**Comment by wjrogers** [comment]
@grumpydev I have been using the build 2.0.0-pre1911 build from your [MyGet feed](https://www.myget.org/gallery/nancyfx) in production for about a year. This build is much newer than even the "clinteastwood" release, and it fixed something important for me; unfortunately, I forget exactly what (maybe a build issue?). I would be happy to see this build promoted to release status.

**Comment by grumpydev** [comment]
Thanks for the comments folks, I will be taking a look Saturday morning (UK time).

**Comment by grumpydev** [comment]
Some packages are still verifying, but I've just pushed v2.0.0 to nuget based off the latest master.

**Comment by m-barthelemy** [comment]
This is awesome, thanks a ton @grumpydev ! Just upgraded 2 of my projects from clinteastwood.
Tests pass and everything seems to be working well.
The only change I had to make was to set `excludeNullValues` in `Nancy.Json.JsonConfiguration`.

**Comment by jchannon** [comment]
Just tested on a brand new netcoreapp 2.2 on osx and al fine!

**Comment by zulq** [comment]
Even though I have used `asp.net-core` to develop and deploy multiple apps in production. I think Nancy should continue thriving as alternative to the official MS Stack. Even though MS open source most stuff  and borrowed/implemented most of the concepts, personally I think there should be viable alternatives and Nancy happens to be a bright example.

**Comment by nkosi23** [comment]
@zulq strongly feeling the same. We use Nancy every day of the week here and if anything, we plan to increase our use of Nancy for example by using it as an embedded HTTP server in various products (databases etc...), in addition to the fact that all our web projects (even new and upcoming ones) are and will continue to be based on Nancy.

Beyond the beauty of its paradigm, its versatility, its lightweight nature, its comprehensive happy-path, and its flexibility, there is also the fact that the source code of Nancy is what I call "simply open source". Certain codebases are so complex that they are for all practical purposes closed-source since there is no way one will ever have the budget to dig into them let alone adjust them to meet his needs. However, Nancy has a small and easy to understand codebase (in addition to being built for extensibility right from the start)

So to me Nancy shouldn't go anywhere. It is done software and does not need to evolve much. I see it as awk and grep, it's not because it is not updated often that it is any less relevant. It is simply that the product has become mature and stable. Just providing my feedback.

**Comment by zulq** [comment]
I have created `.net-core 3 preview` successfully with `ef core` using `npgsql`. Hoping to add more features and do more tests. Those interested [Nancy .Net-Core 3 Demo](https://github.com/zulq/NancyDemo). 

A big catch is that `synchronous` is disable on all servers and one have to configure `kestrel` explicitly to allow `synchronous` code. 
See [AllowSynchronousIO disabled in all servers](https://github.com/aspnet/AspNetCore/issues/7644)
